utils/text_formatting.jsx

```diff
@@ -356,7 +356,8 @@ function autolinkHashtags(text, tokens) {
         const index = tokens.size;
         const alias = `$MM_HASHTAG${index}$`;
 
-        if (text.length < Constants.MIN_HASHTAG_LINK_LENGTH + 1) {
+        const {MinimumHashtagLength} = getConfig(store.getState());
```

Comment:
Instead of accessing the store directly here, could you instead pass the `minimumHashtagLength` as an additional argument to `autolinkHashtags`? That would make testing easier since you can pass it explicitly instead of having to set up the store, and it would also be nice since it wouldn't use the global state. You could then have the `Markdown` component get it from the store [here](https://github.com/mattermost/mattermost-webapp/blob/master/components/markdown/index.js)

Reply:
Also, if we don't use it anywhere else, would you also be able to remove the `MIN_HASHTAG_LINK_LENGTH` constant?

Reply:
> Instead of accessing the store directly here, could you instead pass the `minimumHashtagLength` as an additional argument to `autolinkHashtags`?

It sounds good! But I'm little confusing.
I can pass `minimumHashtagLength` from `Markdown` component, but `TextFormatting.formatText` seems to be called from other components such as SchemaAdminSettings, TextDismissableBar, and so on. Should I fix other components the same way as Markdown component?

And `Renderer.text(txt)` in `utils/markdown/renderer.jsx` calls `TextFormatting.doFormatText`, but it seems not to get `minimumHashtagLength` from store in the same way as `Markdown` component. How do I fix this component?

Reply:
I think it's fine to either disable hashtags or just use the previous value of 3 in other places that call `TextFormatting.formatText`. Realistically, I don't think they should ever contain hashtags.

As for `Renderer.text(txt)`, that passes the formatting options that were previously passed to `formatText` since I don't think we're using that renderer directly. As long as we pass them from the `Markdown` component to `formatText`, they should always be passed from there to `doFormatText`

Reply:
Thank you for clarifying! I've fixed this.